receiver/jaegerreceiver/factory.go

```diff
@@ -0,0 +1,125 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaegerreceiver
+
+// This file implements factory for Jaeger receiver.
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/census-instrumentation/opencensus-service/consumer"
+	"github.com/census-instrumentation/opencensus-service/internal/configmodels"
+	"github.com/census-instrumentation/opencensus-service/internal/factories"
+	"github.com/census-instrumentation/opencensus-service/receiver"
+)
+
+var _ = factories.RegisterReceiverFactory(&receiverFactory{})
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "jaeger"
+
+	// Protocol values.
+	protoThriftHTTP     = "thrift-http"
+	protoThriftTChannel = "thrift-tchannel"
+
+	// Default endpoints to bind to.
+	defaultHTTPBindEndpoint     = "127.0.0.1:14268"
+	defaultTChannelBindEndpoint = "127.0.0.1:14267"
+)
+
+// receiverFactory is the factory for Jaeger receiver.
+type receiverFactory struct {
+}
+
+// Type gets the type of the Receiver config created by this factory.
+func (f *receiverFactory) Type() string {
+	return typeStr
+}
+
+// CustomUnmarshaler returns nil because we don't need custom unmarshaling for this config.
+func (f *receiverFactory) CustomUnmarshaler() factories.CustomUnmarshaler {
+	return nil
+}
+
+// CreateDefaultConfig creates the default configuration for Jaeger receiver.
+func (f *receiverFactory) CreateDefaultConfig() configmodels.Receiver {
+	return &ConfigV2{
+		Protocols: map[string]*configmodels.ReceiverSettings{
+			protoThriftTChannel: {
+				Enabled:  false,
+				Endpoint: defaultTChannelBindEndpoint,
+			},
+			protoThriftHTTP: {
+				Enabled:  false,
+				Endpoint: defaultHTTPBindEndpoint,
+			},
+		},
+	}
+}
+
+// CreateTraceReceiver creates a trace receiver based on provided config.
+func (f *receiverFactory) CreateTraceReceiver(
+	ctx context.Context,
+	cfg configmodels.Receiver,
+	nextConsumer consumer.TraceConsumer,
+) (receiver.TraceReceiver, error) {
+
+	// Convert settings in the source config to Configuration struct
+	// that Jaeger receiver understands.
+
+	rCfg := cfg.(*ConfigV2)
+
+	protoHTTP := rCfg.Protocols[protoThriftHTTP]
+	protoTChannel := rCfg.Protocols[protoThriftHTTP]
+
+	config := Configuration{}
+
+	// Set ports
+	if protoHTTP != nil {
+		config.CollectorHTTPPort = exatractPortFromEndpoint(protoHTTP.Endpoint)
+	}
+	if protoTChannel != nil {
+		config.CollectorHTTPPort = exatractPortFromEndpoint(protoTChannel.Endpoint)
+	}
+
```

Comment:
Should there be a check that at least one of HTTP or TChannel is enabled?

Reply:
Thanks, your comment prompted me to add the config error checks and corresponding tests. Please check again.